src/common/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,98 @@
+"use client";
+
+import {
+  forwardRef,
+  type ReactNode,
+  type ElementRef,
+  type ComponentPropsWithoutRef,
+} from "react";
+import * as DialogPrimitive from "@radix-ui/react-dialog";
+
+import { dialogTheme, type DialogVariants } from "./Dialog.theme";
+import { XCloseIcon } from "@/common/components/CustomIcon";
+
+const DialogRoot = DialogPrimitive.Root;
+
+const DialogTrigger = DialogPrimitive.Trigger;
+
+const DialogPortal = DialogPrimitive.Portal;
+
+const DialogOverlay = forwardRef<
+  ElementRef<typeof DialogPrimitive.Overlay>,
+  ComponentPropsWithoutRef<typeof DialogPrimitive.Overlay>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Overlay ref={ref} className={className} {...props} />
+));
+DialogOverlay.displayName = DialogPrimitive.Overlay.displayName;
+
+export const DialogClose = DialogPrimitive.Close;
+
+export type DialogProps = DialogVariants & {
+  trigger: ReactNode;
+  children: ReactNode;
+  dialogRootProps?: ComponentPropsWithoutRef<typeof DialogRoot>;
+  dialogOverlayProps?: ComponentPropsWithoutRef<typeof DialogOverlay>;
+  dialogTriggerProps?: ComponentPropsWithoutRef<typeof DialogTrigger>;
+  dialogContentProps?: ComponentPropsWithoutRef<typeof DialogPrimitive.Content>;
+  dialogCloseProps?: ComponentPropsWithoutRef<typeof DialogClose>;

```

Comment:
Similar to Popover PR comment, shall we let the consumer add the composition components like trigger and content as children components in Dialog?

Reply:
sure! updated at b61142f